### ISSUE TITLE
Handle WebDAV locks

### ISSUE THREAD

**Issue by cillianderoiste** [ploneintranet/ploneintranet#666]
A user reported they got the following error when trying to delete an object via the standard CMS/barceloneta interface:

```
Module ZPublisher.Publish, line 138, in publish
Module ZPublisher.mapply, line 77, in mapply
Module ZPublisher.Publish, line 48, in call_object
Module z3c.form.form, line 218, in call
Module z3c.form.form, line 213, in update
Module plone.app.z3cform.csrf, line 21, in execute
Module z3c.form.action, line 98, in execute
Module z3c.form.button, line 315, in call
Module z3c.form.button, line 170, in call
Module plone.app.content.browser.actions, line 61, in handle_delete
Module plone.dexterity.content, line 707, in manage_delObjects
Module plone.app.linkintegrity.monkey2, line 19, in manage_delObjects
Module OFS.ObjectManager, line 536, in manage_delObjects
ResourceLockedError: Object "test-delete" is locked via WebDAV
```

I didn't verify this myself yet, but apart from displaying when an item is locked in the barceloneta theme, and providing a more useful message when someone tries to delete a locked item, we probably also need to expose this in the PI theme.

**Comment by ale-rt** [comment]
I think first we need design for this. Is it correct @gyst?

**Comment by mattss** [comment]
Related to https://github.com/ploneintranet/ploneintranet.prototype/issues/91

**Comment by gyst** [comment]
Todo: visually document current (Barceloneta) interactions as base for @cornae

**Comment by gyst** [comment]
Design for PI is already available in the prototype:
http://prototype.ploneintranet.net/projection-material/

It's unclear what exactly the scope of this issue is:
- exposing the lock in PI (see above)
- and/or: exposing the lock in Barceloneta (???)

**Comment by gyst** [comment]
FYI https://github.com/plone/Products.CMFPlone/issues/345

**Comment by pilz** [comment]
this is fixed in master for 1.3
